### ISSUE TITLE
Make k-permutation API more easy to use

### ISSUE THREAD

**Issue by fengertao** [dpaukov/combinatoricslib3#8]
Thanks dpaukov, very good library. I use this library validate my combination/permutation questions, it is easy to use and quick.

One minor issue is I use k-permutation very frequently, while current API is a little verbose
```Java
        Generator.combination(1, 2, 3)
            .simple(2)
            .stream()
            .forEach(combination -> Generator.permutation(combination)
                .simple()
                .forEach(System.out::println));
```

Can you help provide high level API like:
```java
        Generator.permutation(1, 2, 3)
            .simple(2)
            .stream()
            .forEach(System.out::println));
```

Thanks a lot.

**Comment by fengertao** [comment]
Hi, Paukov:

Any comments on the updated PR?

**Comment by fengertao** [comment]
Hi, Paukov:

Last month I have updated PR per our discussion.

Any comments on the updated PR?

BR
Charlie

**Comment by dpaukov** [comment]
This change will be included into the next release version 3.3.2 scheduled on May 2021.
